db/dcrpg/pgblockchain.go

```diff
@@ -5261,61 +5262,183 @@ func (pgb *ChainDB) GetMempoolSSTxDetails(N int) *apitypes.MempoolTicketDetails
 	return &mpTicketDetails
 }
 
-// GetAddressTransactionsRawWithSkip returns an array of apitypes.AddressTxRaw objects
-// representing the raw result of SearchRawTransactionsverbose
-func (pgb *ChainDB) GetAddressTransactionsRawWithSkip(addr string, count int, skip int) []*apitypes.AddressTxRaw {
-	address, err := stdaddr.DecodeAddress(addr, pgb.chainParams)
+// GetAddressTransactionsRawWithSkip returns an array of apitypes.AddressTxRaw
+// objects similar to the raw result of SearchRawTransactionsVerbose, but with
+// fewer Vin details.
+func (pgb *ChainDB) GetAddressTransactionsRawWithSkip(addr string, count, skip int) []*apitypes.AddressTxRaw {
+	_, err := stdaddr.DecodeAddress(addr, pgb.chainParams) // likely redundant with checks in caller
 	if err != nil {
 		log.Infof("Invalid address %s: %v", addr, err)
 		return nil
 	}
+
+	_, height := pgb.BestBlock() // for confirmations
+
 	ctx, cancel := context.WithTimeout(pgb.ctx, 10*time.Second)
 	defer cancel()
-	txs, err := pgb.Client.SearchRawTransactionsVerbose(ctx, address, skip, count, true, true, nil)
+
+	// vins
+	rows, err := pgb.db.QueryContext(ctx, internal.SelectVinsForAddress, addr, count, skip)
```

Comment:
I tried the temporary table in a db txn approach, and what I found was that for many typical addresses, it would actually increase the total time for these queries (e.g. 3 ms -> 13 ms), while for larger addresses it would decrease it a moderate amount (e.g. 600 ms -> 400 ms).

I kept it with the repeated sub-query rather than a temporary table.